### DISCUSSION TITLE
How to orderBy / SortBy products from the API?

### DISCUSSION THREAD

**Discussion by abathellier** [medusajs/medusa#2624] [Q&A] [answered]
I want to sort the products retrieved from the API. For example by `title` or by `updatedAt`. (And this should be the same for all lists)

For instance, a query from the [Saleor](https://demo.saleor.io/graphql/) API: 

```graphql
{
  products(
    first: 5
    channel: "default-channel"
    sortBy: { field: NAME, direction: ASC }
  ) {
    edges {
      node {
        id
        name
        description
      }
    }
  }
}
```

**Comment by adrien2p** [comment, marked as answer]
You can do that in most endpoint using the order query param. If it is not available then we can add it pretty easily normally

**Reply by abathellier** [reply]
I don't think it is available in the product endpoint: https://docs.medusajs.com/api/admin/#tag/Product/operation/GetProducts

And is it possible to sort by direction ASC or DESC with the order query param?

**Reply by adrien2p** [reply]
Yes it is available in the list products handler. You can pass something like order=id for ASC ordering and order=-id for DESC ordering. We need to update the OAS to take that into account in the documentation 👌 the lis product endpoint is automatically ordered by created_at if no order is specified. All endpoint that have been migrated to use the transform middleware allows to pass expand, fields and order. I have created an internal tasks to check that all the oas are up to date 👍

**Reply by abathellier** [reply]
Ok great!

**Reply by abathellier** [reply]
@adrien2p The `order` option doesn't exist on the medusa js client for some endpoints (products.list in this case) : 

```ts
medusa.admin.products.list({
    offset: 0,
    limit: 20,
    order: "-title" // this is not working
  });
```

Could you regenerate the client or something?

**Reply by adrien2p** [reply]
@durdenx can I get you z an issue please

**Reply by abathellier** [reply]
yes https://github.com/medusajs/medusa/issues/2657
